src/io/WKTReader.cpp

```diff
@@ -89,7 +89,7 @@ WKTReader::getCoordinates(StringTokenizer* tokenizer)
         nextToken = getNextCloserOrComma(tokenizer);
     }
 
-    return std::move(coordinates);
+    return coordinates;
```

Comment:
AFAIK no, the problem is that GCC4.8 isn't clever enough to cast CoordinateArraySequence into its base clase, a CoordinateSequence, and `std::move` is a cast so it's doing that for you. Returning a static_cast<function_return_type> should be enough too.

In any case, why complicate the code with unique_ptr instead of using references and return value optimization?

Reply:
`std::unique_ptr<CoordinateSequence>` is not a base class of `std::unique_ptr<CoordinateArraySequence>`, though. The code returns a pointer because GEOS `CoordinateSequences `are always handled as pointers to heap memory because they're polymorphic.

Reply:
As far as I can see, `getCoordinates(tokenizer)` is only used in `readLineStringText` and `readLinearRingText`, none of which need the smart pointer at all. If you change `getCoordinates` to return a `CoordinateSequence` or a `CoordinateArraySequence` (no pointers) you can change `readLineStringText` into:
```
std::unique_ptr<LineString>
WKTReader::readLineStringText(StringTokenizer* tokenizer)
{
    return geometryFactory->createLineString(getCoordinates(tokenizer));
}
```

By adding a constructor of `createLineString^ based on move and not a pointer, which will gain ownership of the coordinate sequence:
```
std::unique_ptr<LineString> createLineString( CoordinateSequence&& coordinates) const;
``` 
Which replaces:
```
std::unique_ptr<LineString> GeometryFactory::createLineString(CoordinateSequence::Ptr && newCoords)
```

That way you remove the need of using smart pointers (which IMO is complicating things quite a bit, as you need multiple std::move calls). Any modern compiler should be able to generate more efficient code by avoiding unnecessary copies / moves.

Reply:
Eventually you need a pointer, because geometries own a pointer to their coordinates. It can't be a value, because `CoordinateSequence` is abstract. Try it if you want?

Reply:
BTW, `readLineStringText` can already be simplified to `return geometryFactory->createLinearRing(getCoordinates(tokenizer));`. The two-line formulation is just the style of whomever wrote this originally.

Reply:
I see the problem now, I didn't realize CoordinateSequence was an abstract class. So it's either pointers or c++17 variants as far as i can see.